exp/modules/modules.go

```diff
@@ -0,0 +1,101 @@
+package modules
+
+import (
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/chanzuckerberg/fogg/config"
+	fogg_hcl "github.com/chanzuckerberg/fogg/exp/hcl"
+	"github.com/chanzuckerberg/go-misc/sets"
+	getter "github.com/hashicorp/go-getter"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+func Run(fs afero.Fs, configFile, path string) error {
+	// figure out which component or account we are talking about
+	conf, err := config.FindAndReadConfig(fs, configFile)
+	if err != nil {
+		return err
+	}
+
+	component, err := conf.PathToComponentType(path)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("component kind %s", component.Kind)
+
+	sources, err := collectModuleSources(path)
+	if err != nil {
+		return err
+	}
+
+	logrus.Debugf("found sources %v", sources)
+
+	for _, s := range sources {
+		fmt.Println(s)
+	}
+
+	return nil
+}
+
+func collectModuleSources(path string) ([]string, error) {
+	references, err := extractLocalModules(path)
+	if err != nil {
+		return nil, err
+	}
+	refNames := references.List()
+	sort.Strings(refNames)
+	return refNames, nil
+}
+
+// extractLocalModules will recursively look for all local module references
+// we assume there are no cycles in this dependency graph
+func extractLocalModules(path string) (sets.StringSet, error) {
+	references := sets.StringSet{}
+
+	// for a given path, find all module references
+	err := fogg_hcl.ForeachBlock(path, func(block *hcl.Block) error {
+		if block.Type == "module" {
+			attrs, _ := block.Body.JustAttributes()
+			for _, a := range attrs {
+				if a.Name == "source" {
+					v, err := a.Expr.Value(nil)
+					if err != nil {
+						return err
+					}
+
+					// we only want local modules, so use go-getter to detect "file:" type
+					str, err2 := getter.Detect(v.AsString(), path, getter.Detectors)
```

Comment:
This is blowing up with references to terraform-aws-modules, which uses a source string that looks like a path, along with a version.

A better way may be this from the Terraform documentation: https://www.terraform.io/docs/language/modules/sources.html#local-paths "A local path must begin with either ./ or ../ to indicate that a local path is intended, to distinguish from a module registry address."

Reply:
@mbarrien can I see an example of what you are talking about? That said searchin for a prefix of `./` or `..` would certainly be easier.

Reply:
It was run on Jess's machine, so I only saw a screenshot of it; need to reproduce on my machine to get you the exact text. But it was crashing processing a module that referred to "terraform-aws-modules/security-group-aws" and trying to find files in a non-existent directory

Reply:
Ah I think I know what's going on. Registry modules are not handled by go-getter (terraform special-cases them further up the stack). So my go-getter approach was not going to work.

I switch to a simpler model, if you want to take a look.